ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1043,168 @@ def test_server_option_with_unreachable_ad(self):
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_gidnumber_not_corresponding_existing_group(self, type):
+        """
+        Test checks that sssd can resolve AD users which
+        contain posix attributes (uidNumber and gidNumber)
+        but there is no group with the corresponding gidNumber
+        and override ad trusted user in default trust view.
+        """
+        posixuser = "testuser2@%s" % self.ad_domain
+        with self.set_idoverrideuser(posixuser, uid="10060",
+                                     gid="10049"):
+            self.mod_idrange_auto_private_group(type)
+            (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+            assert (uid == "10060" and gid == "10049")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_only_uid_number_auto_private_group_default(self, type):
+        """
+        Test checks that posix user with only uidNumber defined
+        and gidNumber not set, auto-private-group
+        is set to false/true/hybrid
+        """
+        posixuser = "testuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        if type == "true":
+            (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+            assert uid == gid
+        else:
+            for host in [self.master, self.clients[0]]:
+                result = host.run_command(['id', posixuser], raiseonerr=False)
+                tasks.assert_error(result, "no such user")
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_auto_private_group_primary_group(self, type):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) and there is primary group
+        with gid number defined.
+        """
+        posixuser = "testuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(type)
+        (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+        assert uid == '10042'
+        if type == "true":
+            assert uid == gid
+        else:
+            assert gid == '10047'
+
+    @pytest.mark.parametrize('type', ['hybrid', 'true', "false"])
+    def test_idoverride_with_auto_private_group(self, type):
+        """
+        Override ad trusted user in default trust view
+        and set auto-private-groups=[hybrid,true,false]
+        and ensure that overridden values takes effect.
+        """
+        posixuser = "testuser@%s" % self.ad_domain
+        with self.set_idoverrideuser(posixuser,
+                                     self.uid_override,
+                                     self.gid_override):
+            self.mod_idrange_auto_private_group(type)
+            (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+            assert(uid == self.uid_override
+                   and gid == self.gid_override)
+            if type == "true":
```

Comment:
The `if type == "true"` check can be removed, because in all the cases the id command will return the user's group in the groups=... section